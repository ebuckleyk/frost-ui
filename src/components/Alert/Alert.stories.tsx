import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Terminal } from 'lucide-react';

import { Alert, AlertDescription, AlertTitle } from './Alert';

function AlertDemo(props) {
  return (
    <Alert {...props} className="max-w-[380px]">
      <Terminal className="h-4 w-4" />
      <AlertTitle>Heads up!</AlertTitle>
      <AlertDescription>You can add components to your app using the cli.</AlertDescription>
    </Alert>
  );
}

type ComponentType = React.ComponentProps<typeof Alert>;
const meta: Meta<ComponentType> = {
  component: Alert,
  render: ({ ...args }) => <AlertDemo {...args} />,
  argTypes: {
    variant: {
      control: { type: 'radio' },
    },
  },
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Demo: Story = {};
