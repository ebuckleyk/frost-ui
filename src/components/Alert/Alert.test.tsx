import * as React from 'react';
import { render } from '@testing-library/react';

import { Alert, AlertDescription, AlertTitle } from './Alert';

const Component = () => {
  return (
    <Alert>
      <AlertTitle>Heads up!</AlertTitle>
      <AlertDescription>You can add components to your app using the cli.</AlertDescription>
    </Alert>
  );
};

describe('Alert', () => {
  it('should render Alert component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
