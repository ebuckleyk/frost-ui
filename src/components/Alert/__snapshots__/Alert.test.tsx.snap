// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > should render Alert component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative w-full rounded-lg border px-4 py-3 text-sm backdrop-blur-md backdrop-saturate-200 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground [&>svg~*]:pl-7 bg-background/60 text-foreground"
        role="alert"
      >
        <h5
          class="mb-1 font-medium leading-none tracking-tight"
        >
          Heads up!
        </h5>
        <div
          class="text-sm [&_p]:leading-relaxed"
        >
          You can add components to your app using the cli.
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative w-full rounded-lg border px-4 py-3 text-sm backdrop-blur-md backdrop-saturate-200 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground [&>svg~*]:pl-7 bg-background/60 text-foreground"
      role="alert"
    >
      <h5
        class="mb-1 font-medium leading-none tracking-tight"
      >
        Heads up!
      </h5>
      <div
        class="text-sm [&_p]:leading-relaxed"
      >
        You can add components to your app using the cli.
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
